internal/datastore/crdb/readwrite.go

```diff
@@ -93,34 +93,34 @@ func (rwt *crdbReadWriteTXN) WriteRelationships(mutations []*v1.RelationshipUpda
 
 	// Process the actual updates
 	for _, mutation := range mutations {
-		rel := mutation.Relationship
-		rwt.addOverlapKey(rel.Resource.ObjectType)
-		rwt.addOverlapKey(rel.Subject.Object.ObjectType)
+		rel := mutation.Tuple
+		rwt.addOverlapKey(rel.ResourceAndRelation.Namespace)
+		rwt.addOverlapKey(rel.Subject.Namespace)
 
 		switch mutation.Operation {
-		case v1.RelationshipUpdate_OPERATION_TOUCH:
+		case core.RelationTupleUpdate_TOUCH:
 			rwt.relCountChange++
 			bulkTouch = bulkTouch.Values(
-				rel.Resource.ObjectType,
-				rel.Resource.ObjectId,
-				rel.Relation,
-				rel.Subject.Object.ObjectType,
-				rel.Subject.Object.ObjectId,
-				stringz.DefaultEmpty(rel.Subject.OptionalRelation, datastore.Ellipsis),
+				rel.ResourceAndRelation.Namespace,
+				rel.ResourceAndRelation.ObjectId,
+				rel.ResourceAndRelation.Relation,
+				rel.Subject.Namespace,
+				rel.Subject.ObjectId,
+				stringz.DefaultEmpty(rel.Subject.Relation, datastore.Ellipsis),
```

Comment:
`rel.Subject.Relation` should never be empty, since it is a `core` type: instead, we should validate it is never given to the datastore empty

Reply:
addressed in https://github.com/authzed/spicedb/pull/771/commits/c38d5a48c8670afefed0afd045c2764fddc5e112